packages/quantic/force-app/main/default/lwc/quanticStandaloneSearchBox/quanticStandaloneSearchBox.js

```diff
@@ -0,0 +1,298 @@
+// @ts-nocheck
+import {LightningElement, api, track} from 'lwc';
+// @ts-ignore
+import {
+  registerComponentForInit,
+  initializeWithHeadless,
+} from 'c/quanticHeadlessLoader';
+import {
+  NavigationMixin,
+  // @ts-ignore
+} from 'lightning/navigation';
+import {
+  STANDALONE_SEARCH_BOX_STORAGE_KEY
+} from 'c/quanticUtils';
+
+import search from '@salesforce/label/c.quantic_Search';
+
+const KEYS = {
+  ENTER: 'Enter',
+  ARROWUP: 'ArrowUp',
+  ARROWDOWN: 'ArrowDown',
+};
+const CLASS_WITH_SUBMIT =
+  'slds-combobox__form-element slds-input-has-icon slds-input-has-icon_right slds-input-has-fixed-addon';
+const CLASS_WITHOUT_SUBMIT =
+  'slds-combobox__form-element slds-input-has-icon slds-input-has-icon_left-right';
+
+export default class QuanticStandaloneSearchBox extends NavigationMixin (
+  LightningElement
+) {
+  
+  labels = {
+    search,
+  };
+
+  // Public props
+  /** @type {number} */
+  @api numberOfSuggestions = 5;
+  /** @type {string} */
+  @api placeholder = `${this.labels.search}...`;
+  /** @type {string} */
+  @api engineId;
+  /** @type {boolean} */
+  @api withoutSubmitButton = false;
+  /** @type {string} */
+  @api redirectUrl = '/global-search/%40uri';
+
+  // Private props
+  /** @type {import("coveo").StandaloneSearchBox} */
+  standaloneSearchBox;
+  /** @type {import("coveo").SearchBox} */
+  searchBox;
+  /** @type {import("coveo").Unsubscribe} */
+  unsubscribe;
+
+  /** @type {import("coveo").StandaloneSearchBoxState} */
+  @track state = {
+    analytics: {
+      cause: '',
+      metadata: null,
+    },
+    redirectTo: null,
+    suggestions: [],
+    value: '',
+    isLoading: false,
+    isLoadingSuggestions: false,
+  };
+
+  /** @type {{key: number, value: string}[]} */
+  get suggestions() {
+    return this.state.suggestions.map((s, index) => ({
+      key: index,
+      rawValue: s.rawValue,
+      value: s.highlightedValue,
+    }));
+  }
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize.bind(this));
+  }
+
+  get actualSearchBox() {
+    if(window.location.href.includes(this.redirectUrl)) {
+      return this.searchBox;
+    }
+    return this.standaloneSearchBox;
+  }
+
+  /**
+   * @param {import("coveo").SearchEngine} engine
+   */
+  @api
+  initialize(engine) {
+    this.searchBox = CoveoHeadless.buildSearchBox(engine, {
+      options: this.searchBoxOptions
+    });
+    this.unsubscribeSearchBox = this.searchBox.subscribe(() => this.updateSearchBoxState());
+
+    this.standaloneSearchBox = CoveoHeadless.buildStandaloneSearchBox(engine, {
+      options: {
+        ...this.searchBoxOptions,
+        redirectionUrl: 'http://placeholder.com',
+      },
+    });
+    this.unsubscribe = this.standaloneSearchBox.subscribe(() =>
+      this.updateStandaloneState()
+    );
+  }
+
+  disconnectedCallback() {
+    if (this.unsubscribe) {
```

Comment:
`disconnectedCallback() {
    this.unsubscribe?.();
  }` 
what do you think?

Reply:
```suggestion
disconnectedCallback() {
  this.unsubscribe?.();
  this.unsubscribeSearchBox?.();
}
```

Reply:
I had no idea we could do that :O